src/application-insights/azext_applicationinsights/_params.py

```diff
@@ -0,0 +1,44 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+# pylint: disable=line-too-long
+from ._validators import validate_applications
+
+
+def load_arguments(self, _):
+    with self.argument_context('monitor app-insights metrics show') as c:
+        c.argument('application', options_list=['--app, -a'], id_part='name', help='GUID, app name, or fully-qualified Azure resource name of Application Insights component. The application GUID may be acquired from the API Access menu item on any Application Insights resource in the Azure portal. If using an application name, please specify resource group.')
+        c.argument('metric', options_list=['--metrics', '-m'], help='The metric to retrieve. May be either a standard AI metric or an application-specific custom metric.')
+        c.argument('timespan', options_list=['--timespan', '-t'], help='The timespan over which to retrieve metric values. This is an ISO8601 time period value. If timespan is omitted, a default time range of `PT12H` ("last 12 hours") is used. The actual timespan that is queried may be adjusted by the server based. In all cases, the actual time span used for the query is included in the response.')
+        c.argument('aggregation', help='The aggregation to use when computing the metric values. To retrieve more than one aggregation at a time, separate them with a comma. If no aggregation is specified, then the default aggregation for the metric is used.')
+        c.argument('interval', help='The time interval to use when retrieving metric values. This is an ISO8601 duration. If interval is omitted, the metric value is aggregated across the entire timespan. If interval is supplied, the server may adjust the interval to a more appropriate size based on the timespan used for the query. In all cases, the actual interval used for the query is included in the response.')
```

Comment:
You should look to other `monitor` commands here. Requiring ISO8601 is generally frowned upon as being user-unfriendly.

Reply:
I see that get_datetime_type returns the ISO8601 version of the input. What exactly is the input that users can provide for this? Does it support the same range of options, e.g. doing durations? I'm thinking of what the equivalent of this would be using your suggestion. 

`az monitor app-insights metrics show -a ace-test -g ace-test -m requests/count --timespan P7D --interval PT1D --segment resultCode`

Reply:
@williexu could you clarify here?

Reply:
@alexeldeib if you take a look at ` az monitor autoscale profile create`, I believe the general pattern for monitor commands is to provide a start/end.

@tjprescott for clarification

Reply:
@williexu @tjprescott I can do this, but it doesn't actually cover all of the scenarios? We would still need some method of describing intervals for segmentation operations. 

I saw autoscale actually has `format: duration` in the swaggers, which I think is almost sufficient. But I was under the impression this was not well-supported in the SDKs and it didn't seem like it uses the full range in the CLI? It's start, end, and recurrence? 

For `--timespan`, we can say period is not necessary and go with start/end. But for `--interval` i'm not sure the same applies? Could you clarify the desired solution there?